apps/nextjs/src/app/page.tsx

```diff
@@ -33,10 +32,21 @@ export default async function HomePage() {
               </div>
             }
           >
-            <PostList posts={posts} />
+            <PostListWrapped />
           </Suspense>
         </div>
       </div>
     </main>
   );
 }
+
+async function PostListWrapped() {
+  const posts = await api.post.all();
+  setQueryData((t) => [t.post.all, undefined], posts); // would be cool if this was automatically put to the cache
```

Comment:
this could be abstracted away by a `createCaller` from `@trpc/react-query` that would automatically set data to a query client cache, then the user would just need to put in `<HydrateClient>` blocks whereever they need to have live, updated data in client components